src/vs/platform/dialogs/common/dialogs.ts

```diff
@@ -18,100 +18,166 @@ import { isLinux, isMacintosh, isWindows } from 'vs/base/common/platform';
 import { IProductService } from 'vs/platform/product/common/productService';
 import { deepClone } from 'vs/base/common/objects';
 
-export interface FileFilter {
-	readonly extensions: string[];
-	readonly name: string;
+export interface IDialogArgs {
+	readonly confirmArgs?: IConfirmDialogArgs;
+	readonly inputArgs?: IInputDialogArgs;
+	readonly promptArgs?: IPromptDialogArgs;
 }
 
-export type DialogType = 'none' | 'info' | 'error' | 'question' | 'warning';
+export interface IBaseDialogOptions {
+	readonly type?: Severity | DialogType;
 
-export interface ICheckbox {
-	readonly label: string;
-	readonly checked?: boolean;
+	readonly title?: string;
+	readonly message: string;
+	readonly detail?: string;
+
+	readonly checkbox?: ICheckbox;
+
+	/**
+	 * Allows to enforce use of custom dialog even in native environments.
+	 */
+	readonly custom?: boolean | ICustomDialogOptions;
 }
 
 export interface IConfirmDialogArgs {
 	readonly confirmation: IConfirmation;
 }
 
-export interface IShowDialogArgs {
-	readonly severity: Severity;
-	readonly message: string;
-	readonly buttons?: string[];
-	readonly options?: IDialogOptions;
+export interface IConfirmation extends IBaseDialogOptions {
+
+	/**
+	 * If not provided, defaults to `Yes`.
+	 */
+	readonly primaryButton?: string;
+
+	/**
+	 * If not provided, defaults to `Cancel`.
+	 */
+	readonly cancelButton?: string;
 }
 
-export interface IInputDialogArgs extends IShowDialogArgs {
-	readonly buttons: string[];
-	readonly inputs: IInput[];
+export interface IConfirmationResult extends ICheckboxResult {
+
+	/**
+	 * Will be true if the dialog was confirmed with the primary button pressed.
+	 */
+	readonly confirmed: boolean;
 }
 
-export interface IDialog {
-	readonly confirmArgs?: IConfirmDialogArgs;
-	readonly showArgs?: IShowDialogArgs;
-	readonly inputArgs?: IInputDialogArgs;
+export interface IInputDialogArgs {
+	readonly input: IInput;
 }
 
-export type IDialogResult = IConfirmationResult | IInputResult | IShowResult;
+export interface IInput extends IConfirmation {
```

Comment:
This is just an interface, so no defaults are being picked here. The buttons are picked differently for confirmation vs input here:

https://github.com/microsoft/vscode/blob/19a5794679636e28c36e40bef33838a72cf86f15/src/vs/platform/dialogs/common/dialogs.ts#L317-L320

Reply:
Sure, I meant the comments in line 49. They only apply to that interface being used directly and not via inheritance here. Obviously the comments can say whatever and the code does something else, but generally I would consider comments on an interface as "part of the contract". Anyways, I am new to the code base so if this does not bother you, that's fine. Just wanted to point it out as a reader of the code.

Reply:
Can you clarify the issue? Is it that you think an input dialog is not a specialized form of a confirm dialog?

Reply:
I think it can be modeled either way. Maybe it is just the location of the comment that is off.

As a user of that API, I would look at the IInput interface and see that it inherits from IConfirmation, and then I would look at what I need to provide and see that I could provide an primaryButton label, and if I don't I get a "yes" button. And then I do that and instead of the Yes button I actually get an Okay button, so the comments on the interface incorrectly describe what the implementation actually does. This could be fixed by changing that comment to say "For confirmation dialogs defaults to "Yes" and for confirmation dialogs to "Okay"". 

Again, it is not the end of the world, but given how much effort you just put into rewriting the entire API, it feels unnecessary for this comment to be incorrect. Of course could also model this by not inheriting so that you have two distinct places where you can document what the respective behavior of leaving out the primaryButton label is, that would be slightly cleaner in my opinion. Again, my opinion, you certainly know better what this entails (if other code relies on the fact that an Input Dialog is a specialization of a Confirmation Dialog).

Reply:
Oh yeah! you have an eye for detail, that can be fixed. Pushed https://github.com/microsoft/vscode/commit/9d236ed32516251b06d73614ee9c526fb5c2e86f